### PR TITLE
Add RemoveFiles endpoint

### PR DESCRIPTION
**Pull request by EAndrey1091**
Add RemoveFiles endpoint

### CODE REVIEW

File: src/WikiService.Broker/WikiService.Broker.csproj [hunk 1]
```diff
@@ -12,4 +12,8 @@
 		<ProjectReference Include="..\WikiService.Models.Db\WikiService.Models.Db.csproj" />
 	</ItemGroup>
 
+	<ItemGroup>
+	  <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.61" />
```
Comment (vorobevamv) [inline src/WikiService.Broker/WikiService.Broker.csproj:16]: actual version is 1.1.63;
do you need it here? it's in "Common.Configurations.targets"

File: src/WikiService.Business/WikiService.Business.csproj [hunk 2]
```diff
@@ -9,4 +9,8 @@
 		<ProjectReference Include="..\WikiService.Validation\WikiService.Validation.csproj" />
 	</ItemGroup>
 
+	<ItemGroup>
+	  <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.61" />
```
Comment (vorobevamv) [inline src/WikiService.Business/WikiService.Business.csproj:13]: in targets

File: src/WikiService.Data.Provider.MsSql.Ef/WikiService.Data.Provider.MsSql.Ef.csproj [hunk 3]
```diff
@@ -6,4 +6,8 @@
 		<ProjectReference Include="..\WikiService.Data.Provider\WikiService.Data.Provider.csproj" />
 	</ItemGroup>
 
+	<ItemGroup>
+	  <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.61" />
```
Comment (vorobevamv) [inline src/WikiService.Data.Provider.MsSql.Ef/WikiService.Data.Provider.MsSql.Ef.csproj:10]: in targets

File: src/WikiService.Data.Provider/WikiService.Data.Provider.csproj [hunk 4]
```diff
@@ -6,4 +6,8 @@
 		<ProjectReference Include="..\WikiService.Models.Db\WikiService.Models.Db.csproj" />
 	</ItemGroup>
 
+	<ItemGroup>
+	  <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.61" />
```
Comment (vorobevamv) [inline src/WikiService.Data.Provider/WikiService.Data.Provider.csproj:10]: in targets

File: src/WikiService.Data/ArticleFileRepository.cs [hunk 5]
```diff
@@ -12,5 +17,29 @@ public class ArticleFileRepository : IArticleFileRepository
     {
       _provider = provider;
     }
+    public async Task<bool> RemoveAsync(List<Guid> filesIds)
+    {
+      bool isRemoveSuccess = false;
+
+      if (filesIds == null)
```
Comment (vorobevamv) [inline src/WikiService.Data/ArticleFileRepository.cs]: better "filesIds is null"

File: src/WikiService.Data/WikiService.Data.csproj [hunk 6]
```diff
@@ -8,4 +8,8 @@
 		<ProjectReference Include="..\WikiService.Models.Dto\WikiService.Models.Dto.csproj" />
 	</ItemGroup>
 
+	<ItemGroup>
+	  <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.61" />
```
Comment (vorobevamv) [inline src/WikiService.Data/WikiService.Data.csproj:12]: in targets

File: src/WikiService.Mappers/WikiService.Mappers.csproj [hunk 7]
```diff
@@ -11,4 +11,8 @@
 	  <Folder Include="Db\" />
 	</ItemGroup>
 
+	<ItemGroup>
+	  <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.61" />
```
Comment (vorobevamv) [inline src/WikiService.Mappers/WikiService.Mappers.csproj:15]: in targets

File: src/WikiService.Models.Db/WikiService.Models.Db.csproj [hunk 8]
```diff
@@ -2,4 +2,8 @@
 
 	<Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
 
+	<ItemGroup>
+	  <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.61" />
```
Comment (vorobevamv) [inline src/WikiService.Models.Db/WikiService.Models.Db.csproj:6]: in targets

File: src/WikiService.Models.Dto/WikiService.Models.Dto.csproj [hunk 9]
```diff
@@ -2,4 +2,8 @@
 
 	<Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
 
+	<ItemGroup>
+	  <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.61" />
```
Comment (vorobevamv) [inline src/WikiService.Models.Dto/WikiService.Models.Dto.csproj:6]: in targets

File: src/WikiService.Validation/WikiService.Validation.csproj [hunk 10]
```diff
@@ -7,4 +7,8 @@
 		<ProjectReference Include="..\WikiService.Models.Dto\WikiService.Models.Dto.csproj" />
 	</ItemGroup>
 
+	<ItemGroup>
+	  <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.61" />
```
Comment (vorobevamv) [inline src/WikiService.Validation/WikiService.Validation.csproj:11]: in targets

File: src/WikiService.Data/ArticleFileRepository.cs [hunk 11]
```diff
@@ -12,5 +17,29 @@ public class ArticleFileRepository : IArticleFileRepository
...
+          _provider.ArticlesFiles
+          .Where(x => filesIds.Contains(x.FileId)));
+
+        await _provider.SaveAsync();
+
+        isRemoveSuccess = true;
+      }
+
+      return isRemoveSuccess;
+    }
```
Comment (vorobevamv) [inline src/WikiService.Data/ArticleFileRepository.cs:35]: may be:
    public async Task<bool> RemoveAsync(List<Guid> filesIds)
    {
      if (filesIds is null)
      {
        return false;
      }

      _provider.ArticlesFiles.RemoveRange(
        _provider.ArticlesFiles
        .Where(x => filesIds.Contains(x.FileId)));

      await _provider.SaveAsync();

      return true;
    }

File: src/WikiService.Business/Commands/File/RemoveFilesCommand.cs [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(RemoveFilesRequest request)
+    {
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveWiki))
+      {
+        return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+      }
+
+      if (!_validator.ValidateCustom(request, out List<string> errors))
```
Comment (vorobevamv) [inline src/WikiService.Business/Commands/File/RemoveFilesCommand.cs]: don't need validation here

File: src/WikiService.Validation/File/Interfaces/IRemoveFilesRequestValidator.cs [hunk 13]
```diff
@@ -0,0 +1,11 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.WikiService.Models.Dto.Requests.File;
+
+namespace LT.DigitalOffice.WikiService.Validation.File.Interfaces
```
Comment (vorobevamv) [inline src/WikiService.Validation/File/Interfaces/IRemoveFilesRequestValidator.cs]: don't need

File: src/WikiService.Validation/File/RemoveFileRequestValidator.cs [hunk 14]
```diff
@@ -0,0 +1,18 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.WikiService.Models.Dto.Requests.File;
+using LT.DigitalOffice.WikiService.Validation.File.Interfaces;
+
+namespace LT.DigitalOffice.WikiService.Validation.File
```
Comment (vorobevamv) [inline src/WikiService.Validation/File/RemoveFileRequestValidator.cs]: don't need validation for this endpoint;
user chooses concrete files; and if somehow there would be non-existing Id in list, repo just wouldn't delete it

File: src/WikiService.Broker/WikiService.Broker.csproj (hunk 1 above)
Comment (EAndrey1091) [reply src/WikiService.Broker/WikiService.Broker.csproj:16]: When I use version 1.1.63, I get error in stranger code (in EditArticleCommand).

File: src/WikiService.Business/WikiService.Business.csproj (hunk 2 above)
Comment (EAndrey1091) [reply src/WikiService.Business/WikiService.Business.csproj:13]: When I use version 1.1.63, I get error in stranger code (in EditArticleCommand).

File: src/WikiService.Data.Provider.MsSql.Ef/WikiService.Data.Provider.MsSql.Ef.csproj (hunk 3 above)
Comment (EAndrey1091) [reply src/WikiService.Data.Provider.MsSql.Ef/WikiService.Data.Provider.MsSql.Ef.csproj:10]: When I use version 1.1.63, I get error in stranger code (in EditArticleCommand).

File: src/WikiService.Data.Provider/WikiService.Data.Provider.csproj (hunk 4 above)
Comment (EAndrey1091) [reply src/WikiService.Data.Provider/WikiService.Data.Provider.csproj:10]: When I use version 1.1.63, I get error in stranger code (in EditArticleCommand).

File: src/WikiService.Data/WikiService.Data.csproj (hunk 6 above)
Comment (EAndrey1091) [reply src/WikiService.Data/WikiService.Data.csproj:12]: When I use version 1.1.63, I get error in stranger code (in EditArticleCommand).

File: src/WikiService.Mappers/WikiService.Mappers.csproj (hunk 7 above)
Comment (EAndrey1091) [reply src/WikiService.Mappers/WikiService.Mappers.csproj:15]: When I use version 1.1.63, I get error in stranger code (in EditArticleCommand).

File: src/WikiService.Models.Db/WikiService.Models.Db.csproj (hunk 8 above)
Comment (EAndrey1091) [reply src/WikiService.Models.Db/WikiService.Models.Db.csproj:6]: When I use version 1.1.63, I get error in stranger code (in EditArticleCommand).

File: src/WikiService.Models.Dto/WikiService.Models.Dto.csproj (hunk 9 above)
Comment (EAndrey1091) [reply src/WikiService.Models.Dto/WikiService.Models.Dto.csproj:6]: When I use version 1.1.63, I get error in stranger code (in EditArticleCommand).

File: src/WikiService.Validation/WikiService.Validation.csproj (hunk 10 above)
Comment (EAndrey1091) [reply src/WikiService.Validation/WikiService.Validation.csproj:11]: When I use version 1.1.63, I get error in stranger code (in EditArticleCommand).

File: src/WikiService.Data/ArticleFileRepository.cs (hunk 11 above)
Comment (EAndrey1091) [reply src/WikiService.Data/ArticleFileRepository.cs:35]: I rewrote the code in another way.

**Pull request closed without merging**